### PR TITLE
Deploy `@checked`

### PR DESCRIPTION
**Pull request by abarth**
This patch adds `@checked` everywhere is needed to remove the
`strong_mode_invalid_method_override` strong mode error.  There's still
one place where we're still ignoring
`strong_mode_invalid_method_override`, but I think that case might be a
bug in the analyzer.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @leafpetersen

File: packages/flutter/lib/src/widgets/unique_widget.dart [hunk 1]
```diff
@@ -19,7 +19,9 @@ abstract class UniqueWidget<T extends State> extends StatefulWidget {
   }
 
   @override
-  T createState();
+  T createState(); //ignore: strong_mode_invalid_method_override
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/unique_widget.dart]: nit: space before "ignore"

File: packages/flutter/lib/src/widgets/unique_widget.dart [hunk 2]
```diff
@@ -19,7 +19,9 @@ abstract class UniqueWidget<T extends State> extends StatefulWidget {
...
+  // TODO(abarth): Having to ignore strong_mode_invalid_method_override here
+  // seems like a bug. We should remove the ignore once we sort out the issue.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/unique_widget.dart]: try the thing i sent by e-mail, but if that doesn't work, file a bug and include the bug# here

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/widgets/unique_widget.dart (hunk 2 above)
Comment (abarth) [inline packages/flutter/lib/src/widgets/unique_widget.dart]: I filed https://github.com/dart-lang/sdk/issues/27546

File: packages/flutter/test/foundation/covariant_templates_test.dart [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+class X {}
+
+class Y extends X {}
+
+class A<U extends X> {
+  U u;
+}
+
+void main() {
+  test('Asignment through a covariant template throws exception', () {
```
Comment (Hixie) [inline packages/flutter/test/foundation/covariant_templates_test.dart]: assignment

File: .analysis_options [hunk 4]
```diff
@@ -21,7 +21,6 @@ analyzer:
     # allow overriding fields (if they use super, ideally...)
     strong_mode_invalid_field_override: ignore
     # allow type narrowing
-    strong_mode_invalid_method_override: ignore
```
Comment (leafpetersen) [inline .analysis_options]: woohoo!

**Pull request merged**